packages/common/src/ui/layout/tables/hooks/useColumns.ts

```diff
@@ -17,34 +18,45 @@ export interface ColumnDefinition<T> {
   sortable?: boolean; // defaults to true
 }
 
-export const useColumns =
-  () =>
-  <T extends Record<string, unknown>>(
-    columns: ColumnDefinition<T>[]
-  ): Column<T>[] => {
-    const t = useTranslation();
-    const formatDate = useFormatDate();
+interface ColumnOptions {
+  useSelectionColumn: boolean;
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export const useColumns = <T extends object>(
+  columnsToMap: ColumnDefinition<T>[],
+  options: ColumnOptions = { useSelectionColumn: false }
```

Comment:
I don't think this is good, but OK for now, maybe - intention wasn't for a `useX` for every column, but the more generic ones use in different places.. 🤷

Reply:
Yup much better
75e26c5